tools/bazar/fields/MapField.php

```diff
@@ -195,7 +203,19 @@ public function formatValuesBeforeSaveIfEditable($entry, bool $isCreation = fals
 
     protected function renderStatic($entry)
     {
-        return "";
+        $output = '';
+        $wiki = $this->getWiki();
+        // the map is only showed on the fullpage entry view, or if $wiki->config['showMapInEntryView'] is set to true
+        if (!empty($this->showMapInEntryView) && $this->showMapInEntryView==='1' && ( $wiki->getPageTag() === $entry['id_fiche'] || (!empty($wiki->config['showMapInEntryView']) && $wiki->config['showMapInEntryView'] === true))) {
```

Comment:
```suggestion
        $params=$this->getService(\Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface::class);
        if (!empty($this->showMapInEntryView)
          && $this->showMapInEntryView==='1' 
          && ( 
             $wiki->getPageTag() === $entry['id_fiche']
             || (
                $params->has('showMapInEntryView')
                && $params->get('showMapInEntryView') === true
              )
            )) {
```
Je fais 2 propositions:
 1. casser la longue ligne pour en faciliter la relecture
 2. ne pas utiliser `wiki->config` mais `params` qui est plus stable car non modifiable par un autre bout du wiki après compilation dans `LoadExtension`
 
 Pour l'utilisation de `params`, il faudrait aussi ajouter dans `tools/bazar/config.yaml` le paramètre `showMapInEntryView`, pour éviter les erreurs et ajouter la ligne `        'showMapInEntryView'` à `bazar_editable_config_params`(toujours dans le même fichier) pour permettre sa modification depuis `?GererConfig`
 
```yaml
  bazar_editable_config_params:
   -'...'
   - 'showMapInEntryView'
```

Reply:
en fait le `showMapInEntryView` est justement prévu pour être changé, le principe est que la carte du point dans une fiche n'apparait que si l'on est sur la fiche dans une page entière appelée (ou son iframe), et le   `showMapInEntryView` permet de pouvoir l'afficher dans d'autres contextes : par exemple dans la vue map, cela n'a pas de sens que dans la popup de la carto, il y ai la carto du marqueur de la fiche, mais pour une vue accordéon ou autre template, cela ferait sens. Il serait donc possible dans le template d'initialiser cette valeur, le temps de l'affichage du template, et d'avoir la feature.
Je ne suis pas super satisfait du mécanisme, mais cela est relativement simple a implémenter et ne change pas le comportement par défaut, faudrait juste trouver ailleurs que dans une variable globale, ou dans la configuration en globale.

Reply:
dans ce cas, il vaut mieux utiliser un service qui stocke cette valeur modifiable et l'interroger pour connaître la valeur. 
Ce sera plus fiable dans le temps et plus facile à debugger.
L'usage de `$GLOBALS` ou de `$wiki->config` en mode ouvert à tout le monde, n'aide vraiment pas à relire le code quand ça bug.

Reply:
et dans ce cas, pas besoin de définir de variable dans `config.yaml`

Reply:
c'est un peu "overkill", je vais plutot voir pour en faire un param de bazarliste

Reply:
c'est un autre moyen auquel je n'avais pas pensé `paramètre de bazarliste` mais effectivement, c'est aussi un moyen fiable (qui demande un peu de réglage pour le voir dans les options après)